pkg/apis/awsproviderconfig/v1alpha1/register.go

```diff
@@ -114,6 +126,22 @@ func (codec *AWSProviderConfigCodec) DecodeProviderStatus(providerStatus *runtim
 	return nil
 }
 
+// ProviderStatusFromMachine gets the machine provider status from the specified machine
+func (codec *AWSProviderConfigCodec) ProviderStatusFromMachine(machine *clusterv1.Machine) (*AWSMachineProviderStatus, error) {
+	status := &AWSMachineProviderStatus{}
+	err := codec.DecodeProviderStatus(machine.Status.ProviderStatus, status)
+	return status, err
+}
+
+// ProviderConfigFromMachine gets the machine provider config from the specified machine
+func (codec *AWSProviderConfigCodec) ProviderConfigFromMachine(machine *clusterv1.Machine) (*AWSMachineProviderConfig, error) {
```

Comment:
In a follow up we need update this https://github.com/openshift/installer/blob/master/pkg/asset/machines/aws/machines.go#L46 to point to the right API `APIVersion: "awsproviderconfig.k8s.io/v1alpha1", Kind:       "AWSMachineProviderConfig",` and then update this func to use DecodeProviderConfig instead of yaml

Reply:
Just did.